NSS-Day-Cohort-66/rare-api-golden-droids
# View All Posts (backend)

updated `posts.py` to have view class and serializer for posts with filtering for specific authorized user

Supported route
`/all_posts`

### Steps to test
1. checkout to main and git fetch
2. switch to zd-all-posts branch
3. restart debugger
4. open Postman
5. use GET request for `http://localhost:8000/posts`
you should receive a 200 status and the following response
```
[
    {
        "id": 5,
        "title": "Example Post 5",
        "rare_user": {
            "id": 2,
            "user": {
                "id": 2,
                "full_name": "Tristan Phifer"
            }
        },
        "category": {
            "id": 2,
            "label": "Cooking"
        },
        "publication_date": "2023-11-24",
        "content": "This is the content of Example Post 5."
    },
    {
        "id": 3,
        "title": "Example Post 3",
        "rare_user": {
            "id": 1,
            "user": {
                "id": 1,
                "full_name": "Carrie Belk"
            }
        },
        "category": {
            "id": 1,
            "label": "Lifestyle"
        },
        "publication_date": "2023-11-22",
        "content": "This is the content of Example Post 3."
    },
    {
        "id": 2,
        "title": "Example Post 2",
        "rare_user": {
            "id": 2,
            "user": {
                "id": 2,
                "full_name": "Tristan Phifer"
            }
        },
        "category": {
            "id": 2,
            "label": "Cooking"
        },
        "publication_date": "2023-11-21",
        "content": "This is the content of Example Post 2."
    },
    {
        "id": 1,
        "title": "Example Post 1",
        "rare_user": {
            "id": 1,
            "user": {
                "id": 1,
                "full_name": "Carrie Belk"
            }
        },
        "category": {
            "id": 1,
            "label": "Lifestyle"
        },
        "publication_date": "2023-11-20",
        "content": "This is the content of Example Post 1."
    }
]
```

6. you can check individual users by doing the following:
GET request for 
`http://localhost:8000/posts?rare_user=current`
use this token 
```
Token fa2eba9be8282d595c997ee5cd49f2ed31f65bed
``` 
this should return all posts for Carrie
```
[
    {
        "id": 3,
        "title": "Example Post 3",
        "rare_user": {
            "id": 1,
            "user": {
                "id": 1,
                "full_name": "Carrie Belk"
            }
        },
        "category": {
            "id": 1,
            "label": "Lifestyle"
        },
        "publication_date": "2023-11-22",
        "content": "This is the content of Example Post 3."
    },
    {
        "id": 1,
        "title": "Example Post 1",
        "rare_user": {
            "id": 1,
            "user": {
                "id": 1,
                "full_name": "Carrie Belk"
            }
        },
        "category": {
            "id": 1,
            "label": "Lifestyle"
        },
        "publication_date": "2023-11-20",
        "content": "This is the content of Example Post 1."
    }
]
```

7. switch to this token 
```
978afa7b76527cc21d76d7b5430ab77f73aa3bff
```
and verify that posts are for Tristan
```
[
    {
        "id": 5,
        "title": "Example Post 5",
        "rare_user": {
            "id": 2,
            "user": {
                "id": 2,
                "full_name": "Tristan Phifer"
            }
        },
        "category": {
            "id": 2,
            "label": "Cooking"
        },
        "publication_date": "2023-11-24",
        "content": "This is the content of Example Post 5."
    },
    {
        "id": 2,
        "title": "Example Post 2",
        "rare_user": {
            "id": 2,
            "user": {
                "id": 2,
                "full_name": "Tristan Phifer"
            }
        },
        "category": {
            "id": 2,
            "label": "Cooking"
        },
        "publication_date": "2023-11-21",
        "content": "This is the content of Example Post 2."
    }
]
```
